### PR TITLE
Add support for init with a BLEDevice

### PR DESCRIPTION
**Pull request by abmantis**
I am developing an Home Assistant integration using this library and this is required for the library to be used in Home Assistant.

I also have a few other PRs comming for the lib to be able to be used in an Home Assistant integration.
I was trying to split them in small PRs for easier review, but I can also make it all in one bigger PR (with all the branch `ha_v1`).

### CODE REVIEW

**Review by newAM — APPROVED** [review]
Thanks for the pull-request!

Happy to support home-assistant usecases!  Let me know if you want any help addressing the CI failures.

File: idasen/__init__.py [hunk 1]
```diff
@@ -83,13 +84,17 @@ class IdasenDesk:
     #: Number of times to retry upon failure to connect.
     RETRY_COUNT: int = 3
 
-    def __init__(self, mac: str, exit_on_fail: bool = False):
+    def __init__(self, address_or_ble_device: Union[BLEDevice, str], exit_on_fail: bool = False):
```
[the remark this answers is not in the document]
Comment (abmantis) [reply idasen/__init__.py:87]: Don't you feel it may be a bit confusing for users of the lib? If not, I am Ok naming it back to `mac`

**Comment by abmantis** [comment]
Thanks for the help! I completely forgot to check the CI jobs and only ran pytest, which was passing so I'll have to check those out.

File: idasen/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (newAM) [reply idasen/__init__.py]: A little bit!  However, I generally prefer to not make breaking changes where possible.

**Review by newAM — APPROVED** [review]
Thanks!

**Pull request merged**